controllers/keystoneapi_controller.go

```diff
@@ -1381,24 +1387,46 @@ func (r *KeystoneAPIReconciler) ensureDB(
 	h *helper.Helper,
 	instance *keystonev1.KeystoneAPI,
 ) (*mariadbv1.Database, ctrl.Result, error) {
+
+	// ensure MariaDBAccount exists.  This account record may be created by
+	// openstack-operator or the cloud operator up front without a specific
+	// MariaDBDatabase configured yet.   Otherwise, a MariaDBAccount CR is
+	// created here with a generated username as well as a secret with
+	// generated password.   The MariaDBAccount is created without being
+	// yet associated with any MariaDBDatabase.
+	_, _, err := mariadbv1.EnsureMariaDBAccount(
+		ctx, h, instance.Spec.DatabaseAccount,
+		instance.Namespace, false, "keystone",
```

Comment:
in a follow up we might want to change where `"keystone"` or `instance.Name` is used for DB and DB CR to be `keystone.DatabaseName`, like e.g. in `NewDatabaseForAccount` and remove finalizer cleanup

Reply:
I haven't decided on a consistent approach for these however in Ironic I have DatabaseName and DatabaseCRName, both constants.

I've seen instance.Name used quite a lot but what's the use case for instance.Name being copied as the database name, and if so why not the CR name?        What's the argument for "the DB name / CR name is constant" vs. "it is dynamic on instance.Name" - we aren't being consistent about this across projects and I think "they are both constants" is likely best

Reply:
obviously not for nova cell DBs, just for the usual case of ironic / cinder / keystone / neutronapi /eetc.   in neutron's case we have to use a constant since apparently they have some kind of dynamic scheme for instance.name

Reply:
the "keystone" name in this particular call is a prefix for creating MariaDB usernames.  it is not ever keyed after this as it's not a complete name.